### PR TITLE
docs: add a configuration section

### PR DESCRIPTION
**Pull request by rjnn**
This is a new page under the 'Overview' section that talks about configuring Materialize. It includes guidance for how to set the now mandatory `--workers` flag.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/2451)

### CODE REVIEW

File: doc/user/overview/configuration.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    weight: 2
+---
+
+All of `materialized`'s command line options can be viewed by running `materialized --help`. This page lists additional considerations and best practices when running Materialize.
+
+## Command line flags
+
+### Data directory
+
+Upon startup `materialized` creates a directory where it persists metadata. By default, this directory is called `mzdata` and is situated in the path where you ran `materialized`. Currently, only metadata is persisted in `mzdata`. You can specify a different directory using the `--data-directory` flag. Upon start, `materialized` checks for an existing data directory, and will reinstall source and view definitions fro it if one is found.
```
Comment (benesch) [inline doc/user/overview/configuration.md]: and is situated in the current working directory of the materialized process.

File: doc/user/overview/configuration.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+When each node is started, it must additionally be told which `--process` it is, from `0` to `processes - 1`.
+
+Note that multi-node Materialize clusters are **not** supported by Materialize, and are not permitted under the free usage BSL license without a separate commercial agreement with Materialize. Moreso, nor are they efficient! An `x1.32xlarge` instance on AWS has 128 VCPUs, and will be superior in every way (reliability, cost, ease-of-use) to a multi-node Materialize cluster with the same total number of VCPUs. It is our performance goal that Materialize under that configuration be able to handle every conceivable streaming workload that you may wish to throw at it.
+
+### Listen Address
+
+By default, `materialized` binds to `0.0.0.0:6875`. This means that Materialize will accept any incoming SQL connection to port 6875 from anywhere. It is the responsibility of the network firewall to limit incoming connections. If you wish to configure `materialized` to only listen to, e.g. localhost connections, you can set this to `localhost:6875`. You can also use this to change the port that Materialize listens on from the default `6875`.
```
Comment (benesch) [inline doc/user/overview/configuration.md]: Perhaps mention that the flag is called `--listen-addr`?

**Comment by benesch** [comment]
Note that the flag is called `-w` or `--threads`, but not `--workers`. I think that's only in your PR description though. Didn't see it in the actual text.

File: doc/user/overview/configuration.md (hunk 2 above)
Comment (rjnn) [reply doc/user/overview/configuration.md]: Done.

File: doc/user/overview/configuration.md (hunk 1 above)
Comment (rjnn) [reply doc/user/overview/configuration.md]: Done.

**Comment by sploiselle** [comment]
This is fine for a stopgap; I'll move this into something more akin to the rest of the docs next week.

**Pull request merged**